lib/signet/oauth_2/client.rb

```diff
@@ -743,7 +748,7 @@ def expires_in
       #   The access token lifetime.
       def expires_in=(new_expires_in)
         if new_expires_in != nil
-          @issued_at = Time.now
+          @issued_at = normalize_timestamp(Time.now)
```

Comment:
One more thing: normalize_timestamp should not be necessary here, because it's a noop for instances of Time.

Reply:
Ah gotcha, should have looked at the implementation.